tests/v2/validation/Jenkinsfile.recurring

```diff
@@ -0,0 +1,310 @@
+                    "${imageName} sh -c \"${workPath}pipeline/releaseupgrade/releaseupgrade.sh\""
+
+                  sh "docker cp ${configGeneratorContainer}:${rootPath}${configsDir}/ ."
+
+                  def configs = sh(script: "ls -1 ./${configsDir}", returnStdout: true).split()
+
+                  for (int i = 0; i < configs.size(); i++) {
+                      def configName = configs[i]
+                      echo "pushing ${configName} to the global variable"
+                      configFiles << "${configName}"
+                  }
+
+                  println "Building the image again with the config files"
+                  sh "./rancher/tests/v2/validation/configure.sh"
+                  sh "docker build . -f ./rancher/tests/v2/validation/Dockerfile.e2e -t ${imageName}"
+              }
+              stage('Deploys clusters') {
+                try {
+                  jobs = [:]
+
+                  configFiles.each {
+                    def configPath = "./${configsDir}/$it"
+                    def absoluteConfigPath = "${rootPath}${configsDir}/$it"
+
+                    def testCase = sh (
+                      script: "./yq '.testCases.provisioningTestCase'  ${configPath}",
+                      returnStdout: true
+                    ).trim()
+
+                    def testPackage = sh (
+                      script: "./yq '.testCases.provisioningTestPackage'  ${configPath}",
+                      returnStdout: true
+                    ).trim()
+
+                    jobs["$it-provisioning"] = { 
+                      stage("Parallel Provisioning: $it"){
+                        def containerName = "${jobName}${env.BUILD_NUMBER}_$it-provisioning"
+                        try {
+                          sh "docker run --volumes-from ${buildTestContainer} --name ${containerName} -t -e CATTLE_TEST_CONFIG=\"${absoluteConfigPath}\" " +
+                        "${imageName} sh -c \"gotestsum --format standard-verbose --packages=\"${testsDir}${testPackage}\" --junitfile ${testResultsOut} -- -tags=validation ${testCase} -timeout=\"60m\" -v\""
+
+                          dir("./${patchedConfigsDir}") {
+                            sh "docker cp ${containerName}:${rootPath}${configsDir}/$it ."
+                          }
+                        } catch (err){
+                          sh "docker stop ${containerName}"
+                          sh "docker rm -v ${containerName}"
+                        }
+
+                        sh "docker stop ${containerName}"
+                        sh "docker rm -v ${containerName}"
+                      }
+                    }
+                  }
+                  parallel jobs
+                } catch(err){
+                    echo "Provisioning tests had failures: " + err
+                }
+              }
+              stage('Run Test Validations') {
```

Comment:
I couldn't really wrap my head around this file, so figure me if this seems ignorant, but is it not possible to utilize the existing Makefile targets/scripts for this? I worry that not being able to specify `make e2e` or something similar will turn this test suite into a QA-only suite. If it's not possible to add those targets/scripts, would it be possible to add `README`s to the appropriate directories so one could figure out how to run it locally, and what the expected modes of operation are?

Reply:
This is to run in our Jenkins pipe, which was the idea since the start. Running this as make e2e is another idea that imo would probably require different infrastructure for running our validation tests as part of a PR process, like the idea you had for our own version of the k8s-ci-robot. What this file does is

1. Creates a rancher instance using corral (right now defaults to the v2.7/v2.8-head
2. Provisions clusters to test against (using the provisioning test regex, so it is actually running our provision test to make sure all for validations pass before testing) It does this in parallel.
3. Runs our validations on the clusters, through our config we run specific tags on specific clusters

This wouldn't run locally this would be a part of our Jenkins test runs.